lib/ngx/ssl/clienthello.lua

```diff
@@ -144,6 +177,46 @@ function _M.get_client_hello_ext_present()
     return nil, ffi_str(errmsg[0])
 end
 
+-- return ciphers_table, err
+-- excluding GREASE ciphers
+function _M.get_client_hello_ciphers()
+    local r = get_request()
+    if not r then
+        error("no request found")
+    end
+
+    local ciphers_table = {} -- table_new won't work because I need to de-GREASE
```

Comment:
Consider placing this line after `if rc == FFI_OK` and initializing with `table.new` with a size hit of 16.